src/Common/Description/GenericDescription.tsx

```diff
@@ -1,327 +1,238 @@
+            setModifiedDescriptionText(descriptionText)
+            setEditDescriptionView(!isEditDescriptionView)
+            setSelectedTab(MD_EDITOR_TAB.WRITE)
         }
-        patchApplicationNote(requestPayload)
-            .then((response) => {
-                if (response.result) {
-                    setDescriptionText(response.result.description)
-                    setDescriptionUpdatedBy(response.result.updatedBy)
-                    const _moment = moment(response.result.updatedOn, 'YYYY-MM-DDTHH:mm:ssZ')
-                    const _date = _moment.isValid()
-                        ? _moment.format(DATE_TIME_FORMATS.TWELVE_HOURS_FORMAT)
-                        : response.result.updatedOn
-                    setDescriptionUpdatedOn(_date)
-                    setModifiedDescriptionText(response.result.description)
-                    // eslint-disable-next-line no-param-reassign
-                    appMetaInfo.note = response.result
-                    toast.success(CLUSTER_DESCRIPTION_UPDATE_MSG)
-                    setEditDescriptionView(true)
-                }
-            })
-            .catch((error) => {
-                showError(error)
-            })
     }
 
-    const updateClusterAbout = (): void => {
+    const handleSave = (): void => {
         const isValidate = validateDescriptionText()
         if (!isValidate) {
             return
         }
-        const requestPayload = {
-            id: descriptionId,
-            identifier: Number(clusterId),
-            description: modifiedDescriptionText,
+        try {
+            updateDescription(modifiedDescriptionText)
+            setEditDescriptionView(true)
+        } catch (error) {
+            showError(error)
         }
-        patchClusterNote(requestPayload)
-            .then((response) => {
-                if (response.result) {
-                    setDescriptionText(response.result.description)
-                    setDescriptionUpdatedBy(response.result.updatedBy)
-                    const _moment = moment(response.result.updatedOn, 'YYYY-MM-DDTHH:mm:ssZ')
-                    const _date = _moment.isValid()
-                        ? _moment.format(DATE_TIME_FORMATS.TWELVE_HOURS_FORMAT)
-                        : response.result.updatedOn
-                    setDescriptionUpdatedOn(_date)
-                    setModifiedDescriptionText(response.result.description)
-                    toast.success(CLUSTER_DESCRIPTION_UPDATE_MSG)
-                    setEditDescriptionView(true)
-                }
-            })
-            .catch((error) => {
-                showError(error)
-            })
     }
 
-    const handleSave = () => {
-        if (isClusterTerminal) {
-            updateClusterAbout()
-        } else {
-            updateApplicationAbout()
+    // TODO: add commandName
+    // eslint-disable-next-line consistent-return
+    const editorCustomIcon = (commandName: string): JSX.Element => {
```

Comment:
Can we make a util say getIconFromCommandName since the Tippy wrapper as well as div.flex is same